### ISSUE TITLE
Transparent Compiler: discrepancy between signatures and source files not reported in diagnostics

### ISSUE THREAD

**Issue by majocha** [dotnet/fsharp#16589]
Basically, diagnostics for not implemented stuff in signatures don't show up.
Not when you edit the implementation, and not in the background solution analysis.

**Comment by 0101** [comment]
Do you have a repro project? Because they show up for me.

![image](https://github.com/dotnet/fsharp/assets/217092/a4199ee3-a10a-4ae1-97db-26e514d19d82)

**Comment by majocha** [comment]
Yes, on this branch: [ct-experiment](https://github.com/majocha/fsharp/tree/ct-experiment)
when I open `BuildGraph.fs`.

**Comment by majocha** [comment]
I did something crazy in the code that causes this. But Switching the Transparent Compiler off causes the diagnostics to show up:
<img width="636" alt="image" src="https://github.com/dotnet/fsharp/assets/1760221/f10e7f8b-fd7c-4a00-b510-ed6f7725b46c">

**Comment by 0101** [comment]
Yeah I can reproduce it there. Will try to figure out what's happening.
